src/catalog/src/transaction.rs

```diff
@@ -281,19 +281,28 @@ impl<'a> Transaction<'a> {
         privileges: Vec<MzAclItem>,
     ) -> Result<DatabaseId, Error> {
         let id = self.get_and_increment_id(DATABASE_ID_ALLOC_KEY.to_string())?;
+        // TODO(parkertimmerman): Support creating databases in the System namespace.
+        let id = DatabaseId::User(id);
+        self.insert_database(id, database_name, owner_id, privileges)?;
+        Ok(id)
+    }
+
+    pub(crate) fn insert_database(
```

Comment:
should we get rid of `fn insert_user_database(...)`? As well as the equiv functions for schema and role?

Reply:
Those are still useful for catalog transactions which (almost) only ever create user objects.